hello/src/hello_backend_plugin.c

```diff
@@ -79,29 +87,46 @@ hello_begin(clicon_handle h, transaction_data td) {
 static int
 hello_end(clicon_handle h, transaction_data td) {
     int *data = transaction_arg(td);
+
     printf("*****hello end*****\n");
     free(data);
     return 0;
 }
 
+/*
+ * Validate that the current XML object is of the form:
+ *   <hello xmlns="urn:example:hello">
+ *     <world/>
+ *   </hello>
+ */
 static int
 find_hello_world(cxobj *vec)
 {
     bool world_found = false, ns_found = false;
     cxobj *c;
     int j;
-    char *ns = NULL;
 
     if (strcmp(xml_name(vec), "hello") != 0)
 	return 0;
+    /*
+     * It would be nice to be able to use nscache_xxx() for this.
```

Comment:
The cache is built on-demand. Yes, you should be able to use such a function. It is not so nice to have this in the example code.
Maybe we could add such a function in the lib if there is nothing appropriate.
The function I think could be used here is xml2ns()